ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitterSpec.scala

```diff
@@ -207,6 +217,91 @@ class TransactionCommitterSpec extends AnyWordSpec with Matchers with MockitoSug
         }
       }
     }
+
+    "setting dedup context" should {
+      val deduplicateUntil = protobuf.Timestamp.newBuilder().setSeconds(30).build()
+      val submissionTime = protobuf.Timestamp.newBuilder().setSeconds(60)
+
+      def buildContextAndTransaction(
+          submitterInfoAugmenter: DamlSubmitterInfo.Builder => DamlSubmitterInfo.Builder
+      ) = {
+
+        val context = createCommitContext(None)
+        context.set(Conversions.configurationStateKey, aDamlConfigurationStateValue)
+        val transactionEntrySummary = DamlTransactionEntrySummary(
+          aDamlTransactionEntry.toBuilder
+            .setSubmitterInfo(
+              submitterInfoAugmenter(
+                DamlSubmitterInfo
+                  .newBuilder()
+              )
+            )
+            .setSubmissionTime(submissionTime)
+            .build()
+        )
+        context -> transactionEntrySummary
+      }
+
+      def contextSetDeduplicateUntil(
```

Comment:
```suggestion
      def deduplicateUntil(
```

Reply:
I would keep the set in the name because this actually modifies the state of the context so it's a mutable function

Reply:
Is the mutation of `Context.get` (about accessed inputs) important in this context though? It looks to me like you just want the resulting dedup until, or am I wrong? If that's the case, having a `set` in the name is confusing IMO.

Reply:
Ah yes, I had the thought that this is the one that sets the value in the context. Renamed